static/app/views/lastKnownRouteContextProvider.tsx

```diff
@@ -1,13 +1,27 @@
-import {createContext, useContext} from 'react';
+import {createContext, useContext, useEffect, useRef} from 'react';
 
 import getRouteStringFromRoutes from 'sentry/utils/getRouteStringFromRoutes';
-import usePrevious from 'sentry/utils/usePrevious';
 import {useRoutes} from 'sentry/utils/useRoutes';
 
 interface Props {
   children: React.ReactNode;
 }
 
+/**
+ * Rewrite usePrevious for our use case, to ignore the not found route.
+ * The previous route can be '/*', which isn't useful for issue grouping.
+ */
+function usePrevious<T>(value: T): T {
```

Comment:
Let's just inline the whole thing, so we don't have to give it a name.

we can leave a comment above it

```
// We could use `usePrevious` here if we didn't need the additional logic to 
// ensure we don't track the not found route
```